### PR TITLE
Main activity with player queue

### PR DESCRIPTION
**Pull request by udiWertheimer**
mokoqueue

What it has:
- MainActivity has both current match and player queue
- Long tap in queue launches choose player activity
- Swiping a player in a queue removes him (but animation is still quirky)

What's left:
- Empty state of queue
- **Fix blinking of item when removing (see gif)**
- Live refresh (notifications?)
- "Join Queue" button
- Tests!
- Queue should show both players

### CODE REVIEW

File: app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        ButterKnife.inject(this, rootView);
+
+        mQueueAdapter = new QueueAdapter(getActivity());
+        mQueueListView.setAdapter(mQueueAdapter);
+        mQueueListView.setDismissCallback(new EnhancedListView.OnDismissCallback() {
+            @Override
+            public EnhancedListView.Undoable onDismiss(final EnhancedListView enhancedListView, int i) {
+                try {
+                    new LeaveQueueOperation().run((QueueItem) enhancedListView.getItemAtPosition(i));
+                    ((QueueAdapter) enhancedListView.getAdapter()).loadObjects();
```
Comment (yonbergman) [inline app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java]: use mQueueAdapter

File: app/build.gradle [hunk 2]
```diff
@@ -24,9 +24,10 @@ android {
 dependencies {
     compile fileTree(dir: 'libs', include: ['*.jar'])
     compile 'com.android.support:appcompat-v7:21.0.3'
-    compile 'com.parse.bolts:bolts-android:1.+'
+    compile 'com.parse.bolts:bolts-android:1.1.4'
```
Comment (giladgo) [inline app/build.gradle:27]: why?

File: app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    static class QueueAdapter extends ParseQueryAdapter<QueueItem>{
+
+        public QueueAdapter(Context context) {
+            super(context, new QueryFactory<QueueItem>() {
+                @Override
+                public ParseQuery<QueueItem> create() {
+                    ParseQuery<QueueItem> query = ParseQuery.getQuery(QueueItem.class);
+                    query.orderByDescending("createdAt");
+                    query.include("player.person");
+                    query.include("player.characterA");
```
Comment (yonbergman) [inline app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java:84]: include charB

**Comment by yonbergman** [comment]
@udiWertheimer also missing displaying 2 characters in queu

File: app/build.gradle (hunk 2 above)
Comment (udiWertheimer) [inline app/build.gradle:27]: Android Studio puts a warning when not specifying exact version

File: app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java (hunk 3 above)
Comment (giladgo) [inline app/src/main/java/com/iic/mokojin/PlayerQueueFragment.java:84]: Don't we want to display both characters? There's a special view for that....

File: app/src/main/java/com/iic/mokojin/operation/LeaveQueueOperation.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import com.iic.mokojin.models.QueueItem;
+import com.parse.ParseCloud;
+import com.parse.ParseException;
+
+import java.util.HashMap;
+
+/**
+ * Created by udi on 3/5/15.
+ */
+public class LeaveQueueOperation {
```
Comment (giladgo) [inline app/src/main/java/com/iic/mokojin/operation/LeaveQueueOperation.java:12]: Add tests
I forgot to add tests to my operations as well :)

File: app/src/main/res/layout/fragment_player_queue.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools" android:layout_width="match_parent"
+    android:layout_height="match_parent" tools:context="com.iic.mokojin.PlayerQueueFragment">
+
+    <de.timroes.android.listview.EnhancedListView
+        android:id="@+id/queue_list_view"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"></de.timroes.android.listview.EnhancedListView>
```
Comment (giladgo) [inline app/src/main/res/layout/fragment_player_queue.xml]: `/>`

File: app/src/main/java/com/iic/mokojin/operation/LeaveQueueOperation.java (hunk 4 above)
Comment (udiWertheimer) [inline app/src/main/java/com/iic/mokojin/operation/LeaveQueueOperation.java:12]: :+1: already on the to do list :)

**Comment by giladgo** [comment]
:tiger2:

**Pull request merged**